src/Rules/Comparison/ImpossibleCheckTypeHelper.php

```diff
@@ -69,12 +68,13 @@ public function findSpecifiedType(
 				$functionName = strtolower((string) $node->name);
 				if ($functionName === 'assert' && $argsCount >= 1) {
 					$arg = $node->getArgs()[0]->value;
-					$assertValues = ($this->treatPhpDocTypesAsCertain ? $scope->getType($arg) : $scope->getNativeType($arg))->getConstantScalarValues();
-					if (count($assertValues) !== 1 || !is_bool($assertValues[0])) {
+					$assertValue = ($this->treatPhpDocTypesAsCertain ? $scope->getType($arg) : $scope->getNativeType($arg))->toBoolean();
+					if (! $assertValue->isTrue()->yes() && ! $assertValue->isFalse()->yes()) {
 						return null;
 					}
 
-					return $assertValues[0];
+					/** @var bool */
+					return $assertValue->getConstantScalarValues()[0];
```

Comment:
You can return `$assertValue->isTrue()->yes()` if you want to avoid the var phpdoc

Reply:
Makes sense!